### PR TITLE
update board.c in k210 bsp

### PR DESCRIPTION
**Pull request by chunyexixiaoyu**
…e ,rt_hw_us_delay this function will be called in unistd.c file

## 拉取/合并请求描述：(PR description)

[
add void rt_hw_us_delay(rt_uint32_t usec) funciton in board.c ，the function will delay for some us ,the function can be called in usleep in unistd.c when using POSIX interface

It has been verified in KD233 development board.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [ ] 本拉取/合并符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

File: bsp/k210/driver/board.c [hunk 1]
```diff
@@ -5,6 +5,7 @@
  *
  * Change Logs:
  * Date           Author       Notes
+ *2021-06-10      xiaoyu       add the function that will delay for some us ,the function can be called in usleep in unistd.c when using POSIX interface
```
Comment (mysterywolf) [inline bsp/k210/driver/board.c:8]: implement rt_hw_us_delay() 即可

**Review by mysterywolf — COMMENTED** [review]
implement rt_hw_us_delay

File: bsp/k210/driver/board.c [hunk 2]
```diff
@@ -117,3 +118,20 @@ void rt_hw_cpu_reset(void)
...
+{
+    rt_uint32_t cycle = read_cycle();
+    rt_uint32_t nop_all = usec * sysctl_clock_get_freq(SYSCTL_CLOCK_CPU) / 1000000UL;
+    while (1)
+    {
+        if(read_cycle() - cycle >= nop_all)
+            break;
+    }
+    return;
+}
```
Comment (mysterywolf) [inline bsp/k210/driver/board.c:137]: 最后一行加一个回车

Comment (chunyexixiaoyu) [reply bsp/k210/driver/board.c:137]: 谢谢我试下

Comment (mysterywolf) [reply bsp/k210/driver/board.c:137]: https://github.com/mysterywolf/formatting

Comment (mysterywolf) [reply bsp/k210/driver/board.c:137]: 第8行 末尾多了一个空格

**Comment by mysterywolf** [comment]
顺便请确认一下是否存在这个问题：https://github.com/RT-Thread/rt-thread/issues/4775

File: bsp/k210/driver/board.c [hunk 3]
```diff
@@ -117,3 +118,21 @@ void rt_hw_cpu_reset(void)
...
+ * @param us the delay time of us
+ */
+void rt_hw_us_delay(rt_uint32_t usec)
+{
+    rt_uint32_t cycle = read_cycle();
+    rt_uint32_t nop_all = usec * sysctl_clock_get_freq(SYSCTL_CLOCK_CPU) / 1000000UL;
+    while (1)
+    {
+       if(read_cycle() - cycle >= nop_all)
+                break;
```
Comment (mysterywolf) [inline bsp/k210/driver/board.c:134]: 此处缩进不对

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/k210/driver/board.c [hunk 4]
```diff
@@ -117,3 +118,21 @@ void rt_hw_cpu_reset(void)
...
+    }
+    return;
```
Comment (mysterywolf) [inline bsp/k210/driver/board.c:136]: 没有必要加return

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Review by mysterywolf — APPROVED** [review]

**Pull request merged**